cmd/clusterctl/pkg/internal/util/yaml.go

```diff
@@ -97,3 +97,33 @@ func FromUnstructured(objs []unstructured.Unstructured) ([]byte, error) {
 
 	return JoinYaml(ret...), nil
 }
+
+// IsClusterResource returns true if the resource kind is cluster wide (not namespaced).
+func IsClusterResource(kind string) bool {
+	return !IsResourceNamespaced(kind)
+}
+
+// IsResourceNamespaced returns true if the resource kind is namespaced.
+func IsResourceNamespaced(kind string) bool {
```

Comment:
I don't think it's part of this PR but the resources here will be subject to change in the future. 
I took a look at `kubectl api-resources` command and noticed it was using [`discoveryClient.ServerPreferredResources`](https://github.com/kubernetes/client-go/blob/0ff5a65499e672956d25228b15354d4083973fe7/discovery/discovery_client.go#L103-L108)

Maybe in the future we could use that if it isn't too complicated to pass in the discovery client.